test/util/genesis/legacy_genesis.go

```diff
@@ -78,6 +79,18 @@ func DocumentLegacyBytes(
 		return nil, fmt.Errorf("adding time_iota_ms field: %w", err)
 	}
 
+	// the version field used to be at this consensus_params.version.app_version
+	bz, err = maps.SetField(bz, "consensus_params.version.app_version", strconv.Itoa(int(cp.Version.App)))
+	if err != nil {
+		return nil, fmt.Errorf("adding app_version field: %w", err)
+	}
+
+	// remove the version from the new location.
+	bz, err = maps.RemoveField(bz, "consensus_params.version.app")
+	if err != nil {
+		return nil, fmt.Errorf("removing version.app field: %w", err)
+	}
```

Comment:
Thanks. 

[no change needed] It was a huge pain that Cosmos SDK v0.50.x broke the format of the genesis file. This has come up a few times. Most recently still in:
- https://github.com/celestiaorg/celestia-app/issues/4780
- https://github.com/celestiaorg/celestia-app/pull/5010